tools/gen_ort_dockerfile.py

```diff
@@ -303,23 +293,31 @@ def dockerfile_for_linux(output_file):
 
 RUN cp /workspace/build/${ONNXRUNTIME_BUILD_CONFIG}/libonnxruntime_providers_openvino.so \
        /opt/onnxruntime/lib && \
-    cp ${INTEL_OPENVINO_DIR}/runtime/lib/intel64/libopenvino.so \
+    cp ${INTEL_OPENVINO_DIR}/runtime/lib/intel64/libopenvino.so.${ONNXRUNTIME_OPENVINO_VERSION} \
        /opt/onnxruntime/lib && \
-    cp ${INTEL_OPENVINO_DIR}/runtime/lib/intel64/libopenvino_c.so \
+    cp ${INTEL_OPENVINO_DIR}/runtime/lib/intel64/libopenvino_c.so.${ONNXRUNTIME_OPENVINO_VERSION} \
        /opt/onnxruntime/lib && \
     cp ${INTEL_OPENVINO_DIR}/runtime/lib/intel64/libopenvino_intel_cpu_plugin.so \
        /opt/onnxruntime/lib && \
-    cp ${INTEL_OPENVINO_DIR}/runtime/lib/intel64/libopenvino_ir_frontend.so \
-       /opt/onnxruntime/lib && \
-    cp ${INTEL_OPENVINO_DIR}/runtime/lib/intel64/libopenvino_onnx_frontend.so \
+    cp ${INTEL_OPENVINO_DIR}/runtime/lib/intel64/libopenvino_ir_frontend.so.${ONNXRUNTIME_OPENVINO_VERSION} \
        /opt/onnxruntime/lib && \
-    cp ${INTEL_OPENVINO_DIR}/runtime/lib/intel64/plugins.xml \
-       /opt/onnxruntime/lib && \
-    cp ${INTEL_OPENVINO_DIR}/runtime/3rdparty/tbb/lib/libtbb.so.2 \
+    cp ${INTEL_OPENVINO_DIR}/runtime/lib/intel64/libopenvino_onnx_frontend.so.${ONNXRUNTIME_OPENVINO_VERSION} \
        /opt/onnxruntime/lib && \
+    cp /usr/lib/x86_64-linux-gnu/libpugixml.so.1 /opt/onnxruntime/lib
+
+RUN OV_SHORT_VERSION=`echo ${ONNXRUNTIME_OPENVINO_VERSION} | awk '{print substr($0,3,2)}'` && \
+    OV_SHORT_VERSION=${OV_SHORT_VERSION}`echo ${ONNXRUNTIME_OPENVINO_VERSION} | awk '{print substr($0,6,1)}'` && \
+    OV_SHORT_VERSION=${OV_SHORT_VERSION}`echo ${ONNXRUNTIME_OPENVINO_VERSION} | awk '{print substr($0,8,1)}'` && \
     (cd /opt/onnxruntime/lib && \
-     chmod a-x * && \
-     ln -sf libtbb.so.2 libtbb.so)
+        chmod a-x * && \
+        ln -s libopenvino.so.${ONNXRUNTIME_OPENVINO_VERSION} libopenvino.so.${OV_SHORT_VERSION} && \
+        ln -s libopenvino.so.${OV_SHORT_VERSION} libopenvino.so && \
```

Comment:
Why is this indirection needed? Why can't we symbolically link the `.so` libs directly?

Reply:
The indirection is how the OV library are linked by default, but I do not think there are any issues linking them directly: [Improve OV short version generation and link so files directly](https://github.com/triton-inference-server/onnxruntime_backend/pull/200/commits/4d7e46a9d3a4efb7d3a716a47bbdf69b3ed961ef)

Reply:
I think I worded my question poorly but your commit shows the intention better :D. My question should have been why do we need `<lib>.so.${OV_SHORT_VERSION}` in the first place, but it seems as though we need all three copies of the file for the compilation. Do I understand this correctly?

Reply:
Yes. We need all three copies to compile.

Reply:
This is how they are linked originally in OV:
```
...
lrwxrwxrwx 1 root root         19 Jul  6 23:44 libopenvino.so -> libopenvino.so.2300
-rw-r--r-- 1 root root  632499720 Jul  6 23:40 libopenvino.so.2023.0.0
lrwxrwxrwx 1 root root         23 Jul  6 23:44 libopenvino.so.2300 -> libopenvino.so.2023.0.0
...
```